File: java/src/com/protobytes/buffers/utils/BuffConverterBase.java
package com.protobytes.buffers.utils;

import com.protobytes.buffers.core.BuffBoolean;
import com.protobytes.buffers.core.BuffByte;
import com.protobytes.buffers.core.BuffBytes;
import com.protobytes.buffers.core.BuffDouble;
import com.protobytes.buffers.core.BuffFloat;
import com.protobytes.buffers.core.BuffInt;
import com.protobytes.buffers.core.BuffLongString;
import com.protobytes.buffers.core.BuffShort;
import com.protobytes.buffers.core.BuffString;
import com.protobytes.buffers.core.BuffUByte;
import com.protobytes.buffers.core.BuffUInt;
import com.protobytes.buffers.core.BuffUShort;
import com.protobytes.buffers.core.commons.IBuffInfo;
import com.protobytes.utils.ByteArray;

/**
 * Buff与值的转换器基类
 * 
 * @author Agua.L
 *
 */
public class BuffConverterBase {
	private static BuffConverterBase _instance;

	public static BuffConverterBase getInstance() {
		if (_instance == null)
			_instance = new BuffConverterBase();
		return _instance;
	}

	public Integer byteFromBuff(String key, IBuffInfo buff) {
		buff.setKey(key);
		Integer target = (Integer) buff.getValue();
		return target;
	}

	public BuffByte byteToBuff(Integer target, String key) {
		BuffByte buff = new BuffByte();
		buff.setKey(key);
		buff.setValue(target);
		return buff;
	}
	
	public Integer ubyteFromBuff(String key, IBuffInfo buff) {
		buff.setKey(key);
		Integer target = (Integer) buff.getValue();
		return target;
	}

	public BuffUByte ubyteToBuff(Integer target, String key) {
		BuffUByte buff = new BuffUByte();
		buff.setKey(key);
		buff.setValue(target);
		return buff;
	}

	public Integer shortFromBuff(String key, IBuffInfo buff) {
		buff.setKey(key);
		Integer target = (Integer) buff.getValue();
		return target;
	}

	public BuffShort shortToBuff(Integer target, String key) {
		BuffShort buff = new BuffShort();
		buff.setKey(key);
		buff.setValue(target);
		return buff;
	}
	
	public Integer ushortFromBuff(String key, IBuffInfo buff) {
		buff.setKey(key);
		Integer target = (Integer) buff.getValue();
		return target;
	}

	public BuffUShort ushortToBuff(Integer target, String key) {
		BuffUShort buff = new BuffUShort();
		buff.setKey(key);
		buff.setValue(target);
		return buff;
	}

	public Integer intFromBuff(String key, IBuffInfo buff) {
		buff.setKey(key);
		Integer target = (Integer) buff.getValue();
		return target;
	}

	public BuffInt intToBuff(Integer target, String key) {
		BuffInt buff = new BuffInt();
		buff.setKey(key);
		buff.setValue(target);
		return buff;
	}
	
	public Long uintFromBuff(String key, IBuffInfo buff) {
		buff.setKey(key);
		Long target = (Long) buff.getValue();
		return target;
	}

	public BuffUInt uintToBuff(Long target, String key) {
		BuffUInt buff = new BuffUInt();
		buff.setKey(key);
		buff.setValue(target);
		return buff;
	}

	public Double floatFromBuff(String key, IBuffInfo buff) {
		buff.setKey(key);
		Double target = (Double) buff.getValue();
		return target;
	}

	public BuffFloat floatToBuff(Double target, String key) {
		BuffFloat buff = new BuffFloat();
		buff.setKey(key);
		buff.setValue(target);
		return buff;
	}

	public Double doubleFromBuff(String key, IBuffInfo buff) {
		buff.setKey(key);
		Double target = (Double) buff.getValue();
		return target;
	}

	public BuffDouble doubleToBuff(Double target, String key) {
		BuffDouble buff = new BuffDouble();
		buff.setKey(key);
		buff.setValue(target);
		return buff;
	}

	public Boolean boolFromBuff(String key, IBuffInfo buff) {
		buff.setKey(key);
		Boolean target = (Boolean) buff.getValue();
		return target;
	}

	public BuffBoolean boolToBuff(Boolean target, String key) {
		BuffBoolean buff = new BuffBoolean();
		buff.setKey(key);
		buff.setValue(target);
		return buff;
	}

	public String stringFromBuff(String key, IBuffInfo buff) {
		buff.setKey(key);
		String target = (String) buff.getValue();
		return target;
	}

	public BuffString stringToBuff(String target, String key) {
		BuffString buff = new BuffString();
		buff.setKey(key);
		buff.setValue(target);
		return buff;
	}

	public String longstringFromBuff(String key, IBuffInfo buff) {
		buff.setKey(key);
		String target = (String) buff.getValue();
		return target;
	}

	public BuffLongString longstringToBuff(String target, String key) {
		BuffLongString buff = new BuffLongString();
		buff.setKey(key);
		buff.setValue(target);
		return buff;
	}

	public ByteArray bytesFromBuff(String key, IBuffInfo buff) {
		buff.setKey(key);
		ByteArray target = (ByteArray) buff.getValue();
		return target;
	}

	public BuffBytes bytesToBuff(ByteArray target, String key) {
		BuffBytes buff = new BuffBytes();
		buff.setKey(key);
		buff.setValue(target);
		return buff;
	}
}

File: dart/lib/com/protobytes/buffers/utils/buff_converter_base.dart
import 'package:proto_bytes/com/protobytes/buffers/core/buffs.dart';
import 'package:proto_bytes/com/protobytes/buffers/core/i_buff_info.dart';
import 'package:proto_bytes/com/protobytes/utils/byte_array.dart';

/// Buff与值的转换器基类
class BuffConverterBase {
  static BuffConverterBase? _instance;

  static BuffConverterBase get instance {
    _instance ??= BuffConverterBase();
    return _instance!;
  }

  int byteFromBuff(String key, IBuffInfo buff) {
    buff.key = key;
    int target = buff.value as int;
    return target;
  }

  BuffByte byteToBuff(int target, String key) {
    BuffByte buff = BuffByte();
    buff.key = key;
    buff.value = target;
    return buff;
  }

  int shortFromBuff(String key, IBuffInfo buff) {
    buff.key = key;
    int target = buff.value as int;
    return target;
  }

  BuffShort shortToBuff(int target, String key) {
    BuffShort buff = BuffShort();
    buff.key = key;
    buff.value = target;
    return buff;
  }

  int intFromBuff(String key, IBuffInfo buff) {
    buff.key = key;
    int target = buff.value as int;
    return target;
  }

  BuffInt intToBuff(int target, String key) {
    BuffInt buff = BuffInt();
    buff.key = key;
    buff.value = target;
    return buff;
  }

  double floatFromBuff(String key, IBuffInfo buff) {
    buff.key = key;
    double target = buff.value as double;
    return target;
  }

  BuffFloat floatToBuff(double target, String key) {
    BuffFloat buff = BuffFloat();
    buff.key = key;
    buff.value = target;
    return buff;
  }

  double doubleFromBuff(String key, IBuffInfo buff) {
    buff.key = key;
    double target = buff.value as double;
    return target;
  }

  BuffDouble doubleToBuff(double target, String key) {
    BuffDouble buff = BuffDouble();
    buff.key = key;
    buff.value = target;
    return buff;
  }

  bool boolFromBuff(String key, IBuffInfo buff) {
    buff.key = key;
    bool target = buff.value as bool;
    return target;
  }

  BuffBoolean boolToBuff(bool target, String key) {
    BuffBoolean buff = BuffBoolean();
    buff.key = key;
    buff.value = target;
    return buff;
  }

  String stringFromBuff(String key, IBuffInfo buff) {
    buff.key = key;
    String target = buff.value as String;
    return target;
  }

  BuffString stringToBuff(String target, String key) {
    BuffString buff = BuffString();
    buff.key = key;
    buff.value = target;
    return buff;
  }

  String longstringFromBuff(String key, IBuffInfo buff) {
    buff.key = key;
    String target = buff.value as String;
    return target;
  }

  BuffLongString longstringToBuff(String target, String key) {
    BuffLongString buff = BuffLongString();
    buff.key = key;
    buff.value = target;
    return buff;
  }

  ByteArray bytesFromBuff(String key, IBuffInfo buff) {
    buff.key = key;
    ByteArray target = buff.value as ByteArray;
    return target;
  }

  BuffBytes bytesToBuff(ByteArray target, String key) {
    BuffBytes buff = BuffBytes();
    buff.key = key;
    buff.value = target;
    return buff;
  }
}
